core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/BenchmarkBinPackingNodeAllocator.java

```diff
@@ -0,0 +1,234 @@
+            }
+        }
+
+        public BinPackingNodeAllocatorService getNodeAllocatorService()
+        {
+            return nodeAllocatorService;
+        }
+
+        private MemoryInfo buildWorkerMemoryInfo(DataSize usedMemory, Map<TaskId, DataSize> taskMemoryUsage)
+        {
+            return new MemoryInfo(
+                    4,
+                    new MemoryPoolInfo(
+                            DataSize.of(64, GIGABYTE).toBytes(),
+                            usedMemory.toBytes(),
+                            0,
+                            ImmutableMap.of(),
+                            ImmutableMap.of(),
+                            ImmutableMap.of(),
+                            taskMemoryUsage.entrySet().stream()
+                                    .collect(toImmutableMap(
+                                            entry -> entry.getKey().toString(),
+                                            entry -> entry.getValue().toBytes())),
+                            ImmutableMap.of()));
+        }
+
+        private void assertAcquired(NodeAllocator.NodeLease lease)
+        {
+            assertEventually(() -> {
+                assertThat(lease.getNode().isCancelled())
+                        .describedAs("node lease cancelled")
+                        .isFalse();
+                assertThat(lease.getNode().isDone())
+                        .describedAs("node lease not acquired")
+                        .isTrue();
+            });
+        }
+
+        private void assertNotAcquired(NodeAllocator.NodeLease lease)
+        {
+            assertThat(lease.getNode().isCancelled())
+                    .describedAs("node lease cancelled")
+                    .isFalse();
+            assertThat(lease.getNode().isDone())
+                    .describedAs("node lease acquired")
+                    .isFalse();
+            // enforce pending acquires processing and check again
+            nodeAllocatorService.processPendingAcquires();
+            assertThat(lease.getNode().isCancelled())
+                    .describedAs("node lease cancelled")
+                    .isFalse();
+            assertThat(lease.getNode().isDone())
+                    .describedAs("node lease acquired")
+                    .isFalse();
+        }
+
+        private static void assertEventually(Runnable assertion)
+        {
+            Assert.assertEventually(
+                    new io.airlift.units.Duration(1000, MILLISECONDS),
+                    new io.airlift.units.Duration(10, MILLISECONDS),
+                    assertion::run);
+        }
+    }
+
+    public static void main(String[] args)
+            throws Exception
+    {
+        // assure the benchmarks are valid before running
+        BenchmarkData data = new BenchmarkData();
+        data.setup();
+        BenchmarkBinPackingNodeAllocator benchmark = new BenchmarkBinPackingNodeAllocator();
+        benchmark.benchmarkProcessPendingAllocations(data);
```

Comment:
This should rather move into a `@Test` method.
There's no necessity to do this in the main method

Reply:
Sure - coppied some pattern - but `@Test` seems nicer